src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -606,11 +607,18 @@ public static SpecialType GetSpecialTypeSafe(this TypeSymbol? type)
             return type is object ? type.SpecialType : SpecialType.None;
         }
 
-        public static bool IsAtLeastAsVisibleAs(this TypeSymbol type, Symbol sym, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        public static unsafe bool IsAtLeastAsVisibleAs(this TypeSymbol type, Symbol sym, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
             CompoundUseSiteInfo<AssemblySymbol> localUseSiteInfo = useSiteInfo;
-            var result = type.VisitType((type1, symbol, unused) => IsTypeLessVisibleThan(type1, symbol, ref localUseSiteInfo), sym,
-                                        canDigThroughNullable: true); // System.Nullable is public
+            var localUseSiteInfoPtr = (nint)Unsafe.AsPointer(ref localUseSiteInfo);
```

Comment:
>var localUseSiteInfoPtr = (nint)Unsafe.AsPointer(ref localUseSiteInfo); [](http://example.com/codeflow?start=12&length=71)

If we decide that we are comfortable going with this `unsafe` gymnastics, we should be able to get rid of the ```localUseSiteInfo``` local. It was added so that we could capture it in a closure.

Reply:
The ref argument isn't guaranteed to be on the stack, so eliminating the local adds new gymnastics of a `fixed` statement (which ends up declaring a pointer to a non-blittable type and gives CS8500).